idaes/apps/uncertainty_propagation/uncertainties.py

```diff
@@ -0,0 +1,248 @@
+        coo_ = sparse.coo_matrix((V,(I,J)))
+    else:
+        coo_ = propagate_results.gradient_c
+
+    Output = namedtuple('Output',['obj', 'theta', 'theta_names', 'cov','gradient_f', 'gradient_c', 'dsdp', 'propagation_c', 'propagation_f','col'])
+    results= Output(obj, theta_out, theta_names, cov, 
+                     propagate_results.gradient_f,
+                     propagate_results.gradient_c, 
+                     propagate_results.dsdp,
+                     coo_,
+                     propagate_results.propagation_f,
+                     propagate_results.col)
+    return results
+
+def propagate_uncertainty(model_uncertain, theta, cov, theta_names, tee=False, solver_options=None):
+    """
+    This function calculates gradient vector, expectation, and variance of the objective function and constraints  of the model for given estimated optimal parameters and covariance matrix of parameters. It calculates error propagation of the objective function and constraints by using gradient vector and covariance matrix.
+    
+    Args:
+        model_uncertain(function or Pyomo ConcreteModel) : Function is a a python Function that generates an instance of the Pyomo model
+        theta(dict)                    : Estimated parameters 
+        cov(numpy.ndarray)             : Covariance matrix of parameters 
+        theta_names(list of strings)   : List of estimated Var names
+        var_dic(dictionary)            : If any original variable contains apostrophes, need an auxiliary dictionary with keys theta_namess without apostrophes, values with apostrophes. e.g) var_dic: {'fs.properties.tau[benzene,toluene]': "fs.properties.tau['benzene','toluene']", 'fs.properties.tau[toluene,benzene]': "fs.properties.tau['toluene','benzene']"} 
+        tee(bool, optional)            : Indicates that ef solver output should be teed
+        solver_options(dict, optional) : Provides options to the solver (also the name of an attribute)
+
+     Returns:
+         tuple   : results object containing the all information including
+            - results.gradient_f(numpy.ndarray)    : Gradient vector of the objective function with respect to the (decision variables, parameters) at the optimal solution
+            - results.gradient_c(numpy.ndarray)    : Gradient vector of the constraints with respect to the (decision variables, parameters) at the optimal solution. Each row contains [column number, row number, and value], colum order follows variable order in col. If no constraint exists, return []
+            - results.dsdp(numpy.ndarray)          : Gradient vector of the (decision variables, parameters) with respect to paramerters (=theta_name). number of rows = len(theta_name), number of columns= len(col)
+            - results.propagation_c(numpy.ndarray) : Error propagation in the constraints (only constraints have theta_name)  e.g) [[constraint iumber1, error1],[constraint number2, error2]]             
```

Comment:
If this is a sparse numpy array, why remove the constraints that do not contain theta? They would just be zero. Because it is a sparse numpy array, it will not need extra memory to store the extra rows. This makes the code simpler.

Reply:
I changed `results.gradient_c` and `results.dsdp`.